agent_testing/Alice.py

```diff
@@ -0,0 +1,108 @@
+"""Alice return route capable agent.
+
+This server is meant to interpret the messages of the
+corresponding Bob client and carry out their requests.
+"""
+
+import hashlib
+import re
+import os
+from aiohttp import web
+from aries_staticagent import StaticConnection, Target, crypto, utils, route
+
+
+def main():
+
+    """Start a server with a static connection."""
+    their_vk, _ = crypto.create_keypair(seed=hashlib.sha256(b"client").digest())
+    conn = StaticConnection.from_seed(
+        hashlib.sha256(b"server").digest(), Target(their_vk=their_vk)
+    )
+
+    my_list = [{"name": "item1", "features": "feature1"}]
+
+    # Make handlers for our three message types.
+
+    # This handler will deal with the "add" message type.
+    @conn.route("https://github.com/dbluhm/agent-testing/tree/main/docs/list/0.1//add")
+    async def add_message_responder(msg, conn):
+
+        new_item = {
+            "Name": msg["content"]["Name"],
+            "Features": msg["content"]["Features"],
+        }
+
+        my_list.append(new_item)
+        response = {
+            "Success!": "Item successfully added to list. "
+            "Item number = " + str(my_list.index(new_item) + 1) + "."
+        }
+
+        await conn.send_async(
+            {
+                "@type": "https://github.com/dbluhm/agent-testing/tree/main/docs/list/0.1/add",
+                "~l10n": {"locale": "en"},
+                "sent_time": utils.timestamp(),
+                "content": "{}".format(response),
```

Comment:
Since we're creating our own protocol here, we have the flexibility to define what attributes are included. Rather than putting the index of the inserted value into the content and reporting success by stringifying a dictionary, let's make this message have the following attributes:
- `@type`: this is required for all DIDComm messages
- `index`: indication of where the item was inserted in the list
- `success`: true or false

In this case as well, sending an `add` message in response to an `add` message seems odd, doesn't it? Let's make the response to an `add` message be a `result` message. We can reuse this message for the responses to both add and delete.